### PR TITLE
Travis CI Setup

### PR DESCRIPTION
**Pull request by DemoXinMC**
This PR sets up Travis CI as discussed in #151 

As requested, it will build PRs for sanity, as well as invoke pushnuget.sh to create NuGet packages when a tag is created in SFML.NET

In addition to providing a .travis.yml, it corrects 9 warnings in the build process with proper XML commenting and an implementation for Transform.GetHashCode().

Beyond this PR, a member of the SFML.NET repository's team will need to setup Travis CI at https://travis-ci.org/ to allow access to the SFML.NET repo, as well as provide a NuGet API Key to an environment variable in Travis named NUGET_API_KEY in order for tagged releases to be pushed to NuGet.  This is a very quick and simple process.

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
Usually not a big fan of combining multiple, unrelated changes in one PR, but since it's about removing warnings, I guess it's okay.

Can you force push the PR or similar, so the now active Travis CI kicks in?

**Comment by DemoXinMC** [comment]
I'm not sure I'm following what you mean? Travis won't build the repo yet because there's no .travis.yml in the master branch

**Comment by eXpl0it3r** [comment]
It will build this PR because this PR does have a .travis.yml in it. 😉

File: SFML.NuGet.props [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <PropertyGroup>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
 
-    <Version>2.5.0</Version>
+    <Version>2.5.1</Version>
```
Comment (eXpl0it3r) [inline SFML.NuGet.props]: I wouldn't increase the version now. Versioning as well as release tagging should be done once a new version is ready for release.

**Pull request merged**